src/types.ts

```diff
@@ -88,8 +93,7 @@ interface RenderRtlReturn<Component extends SetupComponentType> {
 export type RemainingPropsAndTestOverrides<
   ComponentType extends SetupComponentType,
   BaseProps extends Partial<FullProps<ComponentType>>
-> = RemainingProps<ComponentType, BaseProps> &
-  Pick<Partial<FullProps<ComponentType>>, keyof FullProps<ComponentType>>;
+> = RemainingProps<ComponentType, BaseProps> & Partial<FullProps<ComponentType>>;
```

Comment:
Simplified this type while I was in here. `Pick`ing everything from a partial is the same as just the partial 👍